bundles/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/ICloudService.java

```diff
@@ -246,8 +246,16 @@ public boolean validate2faCode(String code) throws IOException, InterruptedExcep
             this.session.post(AUTH_ENDPOINT + "/verify/trusteddevice/securitycode", JsonUtils.toJson(requestBody),
                     headers);
         } catch (ICloudApiResponseException ex) {
-            logger.debug("Code verification failed.", ex);
-            return false;
+            logger.debug("Exception on code verification with HTTP status {}. Verification might still be successful.",
+                    ex.getStatusCode());
+            logger.trace("Exception on code verification with HTTP status {}. Verification might still be successful.",
+                    ex.getStatusCode(), ex);
```

Comment:
I am sorry to request so many changes for something as simple, but your code will log twice when TRACE level is set.
Rather something like:
```
        if (logger.isTraceEnabled()) {
            logger.trace("Exception on code verification with HTTP status {}. Verification might still be successful.",
                    ex.getStatusCode(), ex);
        } else {
            logger.debug("Exception on code verification with HTTP status {}. Verification might still be successful.",
                    ex.getStatusCode());
        }

Reply:
No offense, but as already said I would go with a simple debug log which also logs the stack trace. In my opinion this here is overkill.

Reply:
If you really like the complexity, I could do it. Please decide.

Reply:
Ok, keep only the trace version, please.

Reply:
👍